app/controllers/reservations_controller.rb

```diff
@@ -1,4 +1,19 @@
 class ReservationsController < ApplicationController
+  # Create a reservation for a hall
+  def create
+    if logged_in?
+      @reservation = current_user.reservations.new(hall_id: reservation_params[:hall_id],
+                                                   reserve_date: reservation_params[:reserve_date], status: 'Pending')
+      if @reservation.save
+        render json: { message: 'reservation created', status: :created }
+      else
+        render json: { message: 'reservation not created' }, status: :unprocessable_entity
+      end
+    else
+      render json: { error: 'Please log in to make a reservation' }, status: :unauthorized
+    end
+  end
```

Comment:
- [ ] your code looks awesome and clean 💯  , but I think you forgot to test if the Hall is available for that day or not, we cannot have multiple reservations for the same date, so we need a small method that check if the hall is available for that date or no 

1. if yes we continue with ur logic 
2. if no we should return an error saying the given date is not available